Add support for super-family checks!

felipesanches opened the pull request:
* Update vertical metrics check to consider values across sibling families
* Add a simple & merely informative check that lists detected sibling family directories

## Description
This pull request addresses the problems described at issue #1487

graphicore commented:
@felipe: This scares me! It feels like the kind of thing that makes automation with the dashboard really really hard: which set of sibling-family files are to include (upstream, master, production) for Font Bakery? The dispatcher creates a PR for a family at a time. Please lets discuss this. I'm just posting here as a reminder.

felipesanches commented:
Sure! I am fully aware that we have dashboard concerns to address. I've read once more your comments from a couple years ago here on the issue tracker. I just think that we were suffering from "paralysis by analysis" here. So I moved on and implemented the basic, most simple implementation possible.

Here are the good aspects of this initial implementation:
* It is simple
* It includes one font file lookup method but can be easily extended to add other strategies to find font files
* If needed we can safely add command-line options (or config files) for providing explicit directories where fontbakery should look for sibling families
* The dashboard will have to solve the problem of collecting all files of sibling families in a single container regardless of what we do at the fontbakery-cli level
* The current implementation probably works nicely with @m4rc1e's TravisCI script for google/fonts (I have not tested that yet, but it is designed with that use case in mind)
* The behaviour of these super-family conditions is absolutely equivalent to family-wide conditions when only a single-family is being checked, so we're also safe here.
* This implementation follows your suggestion (from a couple years ago here: https://github.com/googlefonts/fontbakery/issues/1609#issuecomment-335616088) not to mess with the check runner internals and only use the conditions system to deal with the super-family semantics, keeping the check-runner agnostic.

graphicore commented:
Some quick responses. 


> It is simple

It's simplistic

> It includes one font file lookup method but can be easily extended to add other strategies to find font files

That's working, but it leads to many idiosyncratic, different lookup method implementations. If only one method is needed, that's not so bad. How do you extend it? How do you know which strategy to use? Can it be extended externally or does it have to take place in the font bakery code directly?

> I needed we can safely add command-line options ...

Options and configuration is probably all right, a default place can only be in a vendor profile, not in a general implementation.

> The dashboard will have to solve the problem of collecting all files of sibling families in a single container regardless of what we do at the fontbakery-cli level

That's the part that troubles me, it's *not* a "good" point, it's a PITA.

felipesanches commented:
> How do you extend it?

Send a PR.

> Can it be extended externally or does it have to take place in the font bakery code directly?

Send a PR to feed it with directories provided on the command line or config files or any other external sources of info, etc.

>  a default place can only be in a vendor profile, not in a general implementation.

This implementation is indeed in a vendor profile, the `googlefonts` one. And the single font file lookup strategy currently implemented is exactly what the **google/fonts** repo needs.

> That's the part that troubles me, it's not a "good" point, it's a PITA.

It is a necessary pain. We did nothing about checking super-families in the past couple years. The issue needs to be addressed and this was my first step. I don't want to wait a couple more years.
